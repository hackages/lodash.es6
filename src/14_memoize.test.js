import {memoize} from './14_memoize'

describe('memoize', () => {
  it('should compute the value only once', () => {
    const myMock = jest.fn((...args) => args)
    const memoized = memoize(myMock)

    expect(memoized(42)).toMatchSnapshot()
    expect(memoized(42)).toMatchSnapshot()
    expect(myMock).toHaveBeenCalledTimes(1)
  })

  it('should work with multiple arguments', () => {
    const myMock = jest.fn((...args) => args)
    const memoized = memoize(myMock)

    expect(memoized(42, 'hello', true)).toMatchSnapshot()
    expect(memoized(42, 'hello', true)).toMatchSnapshot()
    expect(myMock).toHaveBeenCalledTimes(1)
  })

  it('recompute with new arguments', () => {
    const myMock = jest.fn((...args) => args)
    const memoized = memoize(myMock)

    expect(memoized(42, 'hello', true)).toMatchSnapshot()
    expect(memoized(43)).toMatchSnapshot()
    expect(myMock).toHaveBeenCalledTimes(2)
  })
})
