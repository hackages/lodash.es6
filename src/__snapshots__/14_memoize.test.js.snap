// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`memoize recompute with new arguments 1`] = `
Array [
  42,
  "hello",
  true,
]
`;

exports[`memoize recompute with new arguments 2`] = `
Array [
  43,
]
`;

exports[`memoize should compute the value only once 1`] = `
Array [
  42,
]
`;

exports[`memoize should compute the value only once 2`] = `
Array [
  42,
]
`;

exports[`memoize should work with multiple arguments 1`] = `
Array [
  42,
  "hello",
  true,
]
`;

exports[`memoize should work with multiple arguments 2`] = `
Array [
  42,
  "hello",
  true,
]
`;
